### PR TITLE
AJ-806, AJ-819: more robust loading and error states for WDS left nav

### PR DESCRIPTION
**Pull request by davidangb**
This PR refactors the loading/ready/error states for WDS's left nav.

It uses the [LoadedData](https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/ajax/loaded-data/useLoadedData.ts) model to track None/Loading/Ready/Error status for both the wds proxy url and the wds schema, and then uses those states to determine what to show in the left nav. I've also added a spinner in the left nav when we're still loading.

This PR does NOT attempt to tackle what is displayed in the main body of the page - that will be handled in AJ-817 and AJ-783.

Implementation notes:
* I tried using the `useLoadedData` helper function, but ran into problems with it. This might be from crossing the JS/TS boundary. I might still try to make this work but it was defying me.
* I renamed the `proxyUrlLoaded` and `wdsSchema` variables to `wdsProxyUrl` and `wdsTypes` … mostly this was convenience; by renaming the vars I was able to easily find everywhere they were used.

### CODE REVIEW

File: src/pages/workspaces/workspace/Data.js [hunk 1]
```diff
@@ -569,17 +569,28 @@ const WorkspaceData = _.flow(
...
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
-      }
+      // get the proxy url for this app from Leo
+      setWdsProxyUrl({ status: 'Loading', state: '' })
+      await wdsDataTableProvider.proxyUrlPromise
+        .then(url => {
+          if (!!url) {
+            setWdsProxyUrl({ status: 'Ready', state: url })
```
Comment (jordansafer) [inline src/pages/workspaces/workspace/Data.js:577]: For some history, we initially had a proxyUrl/setProxyUrl in the data page, and shifted to have things inside the WDS provider to encapsulate WDS specific logic there. I like this change as far as making our WDS interactions more robust, and thinking about if there's a way to encapsulate these WDS interactions in the WDS provider.

**Review by jordansafer — APPROVED** [review]

File: src/pages/workspaces/workspace/Data.js [hunk 2]
```diff
@@ -487,20 +487,20 @@ const WorkspaceData = _.flow(
...
-  const [wdsSchemaError, setWdsSchemaError] = useState()
   const [sidebarWidth, setSidebarWidth] = useState(280)
   const [activeCrossTableTextFilter, setActiveCrossTableTextFilter] = useState('')
   const [crossTableResultCounts, setCrossTableResultCounts] = useState({})
   const [crossTableSearchInProgress, setCrossTableSearchInProgress] = useState(false)
   const [showDataTableVersionHistory, setShowDataTableVersionHistory] = useState({}) // { [entityType: string]: boolean }
-  const [proxyUrlLoaded, setProxyUrlLoaded] = useState(false)
+
+  const [wdsProxyUrl, setWdsProxyUrl] = useState({ status: 'None', state: '' })
+  const [wdsTypes, setWdsTypes] = useState({ status: 'None', state: [] })
```
Comment (aaronkanzer) [inline src/pages/workspaces/workspace/Data.js:503]: Mini nit, but what exactly would the different `Types` of WDS be?

File: src/pages/workspaces/workspace/Data.js (hunk 1 above)
Comment (davidangb) [reply src/pages/workspaces/workspace/Data.js:577]: hm, yeah - I could see moving all this logic into `WdsDataTableProvider` and giving it a `refresh()` or `initialize()` method, and also giving it `isReady` / `isLoading` / `isError` member variables. The tricky part might be that `WdsDataTableProvider` doesn't have any React state to work with.

File: src/pages/workspaces/workspace/Data.js (hunk 2 above)
Comment (davidangb) [reply src/pages/workspaces/workspace/Data.js:503]: these are the WDS record types. `wdsTypes` holds the result of WDS's `describeAllRecordTypes` API.

**Review by aaronkanzer — APPROVED** [review]

**Review by calypsomatic — APPROVED** [review]

**Pull request merged**